src/main/java/dk/cs/aau/huppaal/presentations/TagPresentation.java

```diff
@@ -95,103 +90,118 @@ private void initializeTextAid() {
-
-        final int padding = 0;
-
         label.layoutBoundsProperty().addListener((obs, oldBounds, newBounds) -> {
-            double newWidth = Math.max(newBounds.getWidth(), 10);
-            final double res = GRID_SIZE * 2 - (newWidth % (GRID_SIZE * 2));
-            newWidth += res;
-
-            textField.setMinWidth(newWidth);
-            textField.setMaxWidth(newWidth);
-
-            l2.setX(newWidth + padding);
-            l3.setX(newWidth + padding);
-
-            setMinWidth(newWidth + padding);
-            setMaxWidth(newWidth + padding);
-
-            textField.setMinHeight(TAG_HEIGHT);
-            textField.setMaxHeight(TAG_HEIGHT);
-
-            textField.focusedProperty().addListener((observable, oldFocused, newFocused) -> {
+            var newWidth = Math.max(newBounds.getWidth(), 10);
+            var newHeight = Math.max(newBounds.getHeight(), TAG_HEIGHT);
+            var resX = GRID_SIZE * 2 - (newWidth % (GRID_SIZE * 2));
+            var resY = GRID_SIZE * 2 - (newHeight % (GRID_SIZE * 2));
+            newWidth += resX;
+            newHeight += resY;
+
+            text.setMinWidth(newWidth);
+            text.setPrefWidth(newWidth);
+            text.setMinHeight(TAG_HEIGHT);
+            text.setPrefHeight(TAG_HEIGHT);
+
+            l2.setX(newWidth);
+            l3.setX(newWidth);
+            l3.setY(newHeight);
+            l4.setY(newHeight);
+
+            setMinWidth(newWidth);
+            setMaxWidth(newWidth);
+
+            text.focusedProperty().addListener((observable, oldFocused, newFocused) -> {
                 if (newFocused) {
                     shape.setTranslateY(2);
-                    textField.setTranslateY(2);
+                    text.setTranslateY(2);
                 }
             });
 
             if (getWidth() >= 1000) {
                 setWidth(newWidth);
-                setHeight(TAG_HEIGHT);
+                setHeight(newHeight);
                 shape.setTranslateY(-1);
-                textField.setTranslateY(-1);
+                text.setTranslateY(-1);
             }
 
             // Fixes the jumping of the shape when the text field is empty
-            if (textField.getText().isEmpty()) {
+            if (text.getText().isEmpty()) {
                 shape.setLayoutX(0);
             }
         });
 
-        label.textProperty().bind(new When(textField.textProperty().isNotEmpty()).then(textField.textProperty()).otherwise(textField.promptTextProperty()));
+        label.textProperty().bind(new When(text.textProperty().isNotEmpty()).then(text.textProperty()).otherwise(text.promptTextProperty()));
+    }
 
+    private void stayWithinParent(TextField textField) {
+        Node parent;
+        if(getParent() instanceof NailPresentation){
+            //Get the correct parent for guards, selects, synchronizations, and updates
+            parent = getParent();
+        } else {
+            //Gte the correct parent for location names
```

Comment:
Gte -> Get

Reply:
Fixed in 46644529eb4b7362b9d716d691bd164275d0c1e8